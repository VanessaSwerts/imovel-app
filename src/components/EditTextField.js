import React from 'react'
import { View, StyleSheet, Text, TextInput } from 'react-native'

import colors from '../constants/colors.json'

export default function EditTextField({ label, text, keyboardType }) {
  return (
    <View style={styles.inputContainer}>

      <View style={styles.infoContainer} >
        <Text style={styles.label} >{label}</Text>
        <TextInput style={styles.infoText} keyboardType={keyboardType}
          placeholder={text}
          placeholderTextColor={colors["blue"]}
        />
      </View>

    </View>
  )
}

const styles = StyleSheet.create({
  inputContainer: {
    backgroundColor: "#FFF",
    alignSelf: 'stretch',
    height: 46,
    borderWidth: 1,
    borderColor: colors["blue"],
    borderRadius: 24,
    marginTop: 15,
    paddingHorizontal: 12,
    flexDirection: "row",
    shadowOpacity: 0.2,
    shadowRadius: 2,
    shadowOffset: {
      height: 0,
      width: 0,
    },
    elevation: 2,
  },

  infoContainer: {
    paddingHorizontal: 5,
    alignSelf: 'center',
    flexDirection: 'row',
  },

  label: {
    fontWeight: '600',
    color: colors["blue"],
  },  
})