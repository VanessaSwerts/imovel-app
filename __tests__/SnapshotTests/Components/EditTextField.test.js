import React from 'react'
import { act, create } from 'react-test-renderer'
import EditTextField from '../../../src/components/EditTextField'

describe('<EditTextField />', () => {

  it('render correctly with edit equals false', () => {
    act(() => {
      tree = create(
        <EditTextField
          label={'Celular: '}
          text={'(35) 99988-7766'}
          keyboardType={'phone-pad'}
          edit={false}
        />
      )
    })

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('render correctly with edit equals true', () => {
    act(() => {
      tree = create(
        <EditTextField
          label={'Celular: '}
          text={'(35) 99988-7766'}
          keyboardType={'phone-pad'}
          edit={true}
        />
      )
    })

    expect(tree.toJSON()).toMatchSnapshot()
  })
})