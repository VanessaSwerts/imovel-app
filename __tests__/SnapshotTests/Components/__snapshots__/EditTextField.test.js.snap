// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditTextField /> render correctly with edit equals false 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "borderColor": "#14213D",
      "borderRadius": 24,
      "borderWidth": 1,
      "elevation": 2,
      "flexDirection": "row",
      "height": 46,
      "marginTop": 25,
      "paddingHorizontal": 12,
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "paddingHorizontal": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#14213D",
          "fontWeight": "600",
        }
      }
    >
      Celular: 
    </Text>
    <Text
      style={
        Object {
          "color": "#14213D",
        }
      }
    >
      (35) 99988-7766
    </Text>
  </View>
</View>
`;

exports[`<EditTextField /> render correctly with edit equals true 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "borderColor": "#14213D",
      "borderRadius": 24,
      "borderWidth": 1,
      "elevation": 2,
      "flexDirection": "row",
      "height": 46,
      "marginTop": 25,
      "paddingHorizontal": 12,
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="phone-pad"
    placeholder="(35) 99988-7766"
    placeholderTextColor="#14213D"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#14213D",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
